wls-broadcast-service/src/test/java/de/muenchen/oss/wahllokalsystem/security/BroadcastSecurityTest.java

```diff
@@ -0,0 +1,89 @@
+package de.muenchen.oss.wahllokalsystem.security;
+
+import static de.muenchen.oss.wahllokalsystem.TestConstants.SPRING_TEST_PROFILE;
+import de.muenchen.oss.wahllokalsystem.MicroServiceApplication;
+import de.muenchen.oss.wahllokalsystem.TestConstants;
+import de.muenchen.oss.wahllokalsystem.service.BroadcastService;
+import de.muenchen.oss.wahllokalsystem.utils.UsernamePasswordAuthenticationToken;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.core.authority.AuthorityUtils;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.context.ActiveProfiles;
+
+@SpringBootTest(
+        classes = { MicroServiceApplication.class },
+        webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+        properties = {
+                "spring.datasource.url=jdbc:h2:mem:wahllokalsystem;DB_CLOSE_ON_EXIT=FALSE",
+                "refarch.gracefulshutdown.pre-wait-seconds=0",
+                "server.port=0"
+        }
+)
+@ActiveProfiles(profiles = { SPRING_TEST_PROFILE })
+public class BroadcastSecurityTest {
+
+    @Autowired
+    BroadcastService broadcast_S;
+
+    @BeforeEach
+    public void setUp() {
+        Assertions.assertThat(broadcast_S).isNotNull();
+        SecurityContextHolder.clearContext();
+    }
+
+    @Test
+    public void broadcastAccessDeniedTest() {
+
+        SecurityContextHolder.getContext().setAuthentication(
+                new UsernamePasswordAuthenticationToken(TestConstants.TESTUSER, TestConstants.TESTPASSWORD, AuthorityUtils.createAuthorityList("ROLE_DUMMY")));
+        Exception thrownException = null;
+        try {
+            //noinspection DataFlowIssue
+            broadcast_S.broadcast(null);
+        } catch (Exception e) {
+            thrownException = e;
+        }
+        Assertions.assertThat(thrownException)
+                .isNotNull()
+                .isInstanceOf(AccessDeniedException.class)
+                .hasMessageStartingWith("Access Denied");
```

Comment:
Macht den Code aus meiner Sicht kompakter und gut lesbar. Kann auch auf die nachfolgenden Tests übertragen werden.

```suggestion
        Assertions.assertThatExceptionOfType(AccessDeniedException.class).isThrownBy(() -> broadcast_S.broadcast(null))
                .withMessageStartingWith("Access Denied");
```

Reply:
resolved in a21331ecf44ebbe1a1d381f99b88942e2292b5d6